sql/core/src/main/scala/org/apache/spark/sql/execution/command/createDataSourceTables.scala

```diff
@@ -204,13 +194,21 @@ case class CreateDataSourceTableAsSelectCommand(
       case None => data
     }
 
+    val tableWithPath = if (table.tableType == CatalogTableType.MANAGED) {
+      table.withNewStorage(
+        locationUri = Some(sessionState.catalog.defaultTablePath(table.identifier)))
```

Comment:
This pattern:
```
 val tableWithPath = if (table.tableType == CatalogTableType.MANAGED) {
   table.withNewStorage(
     locationUri = Some(sessionState.catalog.defaultTablePath(table.identifier)))
 ```
appears twice, may be we should create a util method in class CatalogTable?